TestRemoveContainer flakes on container log file removal

https://github.com/kubernetes/kubernetes/pull/93333 modified container log file removal in a way that made TestRemoveContainer flaky

```
=== RUN   TestRemoveContainer
    kuberuntime_container_test.go:84: 
        	Error Trace:	kuberuntime_container_test.go:84
        	Error:      	Not equal: 
        	            	expected: []string{"/var/log/pods/new_bar_12345678/foo/0.log", "/var/log/pods/new_bar_12345678/foo/0.log.20060102-150405", "/var/log/containers/bar_new_foo-bar_new_12345678_0_foo_0.log"}
        	            	actual  : []string{"/var/log/pods/new_bar_12345678/foo/0.log.20060102-150405", "/var/log/pods/new_bar_12345678/foo/0.log", "/var/log/containers/bar_new_foo-bar_new_12345678_0_foo_0.log"}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,4 +1,4 @@
        	            	 ([]string) (len=3) {
        	            	+ (string) (len=56) "/var/log/pods/new_bar_12345678/foo/0.log.20060102-150405",
        	            	  (string) (len=40) "/var/log/pods/new_bar_12345678/foo/0.log",
        	            	- (string) (len=56) "/var/log/pods/new_bar_12345678/foo/0.log.20060102-150405",
        	            	  (string) (len=60) "/var/log/containers/bar_new_foo-bar_new_12345678_0_foo_0.log"
        	Test:       	TestRemoveContainer
--- FAIL: TestRemoveContainer (0.00s)
```

See https://storage.googleapis.com/k8s-gubernator/triage/index.html?pr=1&test=TestRemoveContainer

Note that the flake is masked by the fact that the unit test job will retry tests up to 3 times. With [that disabled](https://github.com/kubernetes/kubernetes/pull/93605), this flake is quite pronounced: https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/93605/pull-kubernetes-bazel-test/1297588838339710977

A fix for the flake is in https://github.com/kubernetes/kubernetes/pull/94185, but it's unclear to me whether it is correct that the removal order is non-deterministic.

/assign @piosz @dchen1107 
/sig node
/milestone v1.19
/priority critical-urgent

--- a/pkg/kubelet/kuberuntime/kuberuntime_container_test.go
+++ b/pkg/kubelet/kuberuntime/kuberuntime_container_test.go
@@ -79,9 +79,12 @@ func TestRemoveContainer(t *testing.T) {
 
 	err = m.removeContainer(containerID)
 	assert.NoError(t, err)
-	// Verify container log is removed
 
-	assert.Equal(t, []string{expectedContainerLogPath, expectedContainerLogPathRotated, expectedContainerLogSymlink}, fakeOS.Removes)
+	// Verify container log is removed.
+	// We could not predict the order of `fakeOS.Removes`, so we use `assert.ElementsMatch` here.
+	assert.ElementsMatch(t,
+		[]string{expectedContainerLogSymlink, expectedContainerLogPath, expectedContainerLogPathRotated},
+		fakeOS.Removes)
 	// Verify container is removed
 	assert.Contains(t, fakeRuntime.Called, "RemoveContainer")
 	containers, err := fakeRuntime.ListContainers(&runtimeapi.ContainerFilter{Id: containerID})

